src/cpp/session/modules/SessionFind.cpp

```diff
@@ -1060,12 +1086,33 @@ core::Error runGrepOperation(const GrepOptions& grepOptions, const ReplaceOption
       cmd << "--include=" + filePattern.getString();
    }
 
-   cmd << shell_utils::EscapeFilesOnly << "--" << shell_utils::EscapeAll;
-
-
    // Filepaths received from the client will be UTF-8 encoded;
    // convert to system encoding here.
    FilePath dirPath = module_context::resolveAliasedPath(grepOptions.directory);
+   for (json::Value filePattern : grepOptions.excludeFilePatterns)
+   {
+      // to retrieve the git exclusions, we run a git command and parse the output
+      if (filePattern.getString().compare("gitExclusions") == 0)
+      {
+         std::string excludeGitExclusion;
+         error = createGitExclusionString(dirPath, &excludeGitExclusion);
+         if (error)
+            return error;
+
+         std::istringstream stream(excludeGitExclusion);
+         std::vector<std::string> results((std::istream_iterator<std::string>(stream)),
+            std::istream_iterator<std::string>());
+         for (std::string filePattern : results)
+         {
+            filePattern.insert(0, dirPath.getAbsolutePath() + "/");
+            cmd << "--exclude=" + filePattern;
```

Comment:
`ls-files` can return an unbounded amount of output, but most operating systems have a pretty small bound on the size of a shell command, which is a recipe for trouble here since we're trying to add every line to the command. This could cause us to fail when there are a lot of ignored files with long paths.

Reply:
I remove the `ls-files` code and rewrote `grep` with `git grep`